iceoryx_posh/test/moduletests/test_roudi_cmd_line_parser.cpp

```diff
@@ -0,0 +1,429 @@
+{
+    constexpr uint8_t numberOfArgs{3U};
+    char* args[numberOfArgs];
+    char appName[] = "./foo";
+    char option[] = "-k";
+    char value[] = "42";
+    args[0] = &appName[0];
+    args[1] = &option[0];
+    args[2] = &value[0];
+
+    iox::config::CmdLineParser sut;
+    auto result = sut.parse(numberOfArgs, args);
+
+    EXPECT_THAT(result.has_error(), Eq(false));
+    EXPECT_THAT(result.value().processKillDelay, Eq(Duration::seconds(42)));
+}
+
+TEST_F(CmdLineParser_test, KillDelayOptionOutOfBoundsLeadsToProgrammNotRunning)
+{
+    constexpr uint8_t numberOfArgs{3U};
+    char* args[numberOfArgs];
+    char appName[] = "./foo";
+    char option[] = "--kill-delay";
+    char value[] = "4294967296"; // MAX_PROCESS_KILL_DELAY + 1
+    args[0] = &appName[0];
+    args[1] = &option[0];
+    args[2] = &value[0];
+
+    iox::config::CmdLineParser sut;
+    auto result = sut.parse(numberOfArgs, args);
+
+    EXPECT_THAT(result.has_error(), Eq(false));
+    EXPECT_THAT(result.value().run, Eq(false));
+}
+
+TEST_F(CmdLineParser_test, CompatibilityLevelOptionsLeadToCorrectCompatibilityLevel)
+{
+    constexpr uint8_t numberOfArgs{3U};
+    CompatibilityCheckLevel loglevelArray[] = {CompatibilityCheckLevel::OFF,
+                                               CompatibilityCheckLevel::MAJOR,
+                                               CompatibilityCheckLevel::MINOR,
+                                               CompatibilityCheckLevel::PATCH,
+                                               CompatibilityCheckLevel::COMMIT_ID,
+                                               CompatibilityCheckLevel::BUILD_DATE};
+    char* args[numberOfArgs];
+    char appName[] = "./foo";
+    char optionArray[][20] = {"-x", "--compatibility"};
+    char valueArray[][10] = {"off", "major", "minor", "patch", "commitId", "buildDate"};
+    args[0] = &appName[0];
+
+    for (auto optionValue : optionArray)
+    {
+        args[1] = optionValue;
+        uint8_t i{0U};
+        for (auto expectedValue : loglevelArray)
+        {
+            args[2] = valueArray[i];
+            testCompatibilityLevel(numberOfArgs, args, expectedValue);
+            i++;
+        }
+    }
+}
+
+TEST_F(CmdLineParser_test, WrongCompatibilityLevelOptionLeadsToProgrammNotRunning)
+{
+    constexpr uint8_t numberOfArgs{3U};
+    char* args[numberOfArgs];
+    char appName[] = "./foo";
+    char option[] = "-x";
+    char wrongValue[] = "AmyPond";
+    args[0] = &appName[0];
+    args[1] = &option[0];
+    args[2] = &wrongValue[0];
+
+    iox::config::CmdLineParser sut;
+    auto result = sut.parse(numberOfArgs, args);
+
+    EXPECT_THAT(result.has_error(), Eq(false));
+    EXPECT_THAT(result.value().run, Eq(false));
+}
+
+TEST_F(CmdLineParser_test, UniqueIdLongOptionLeadsToCorrectUniqueId)
+{
+    constexpr uint8_t numberOfArgs{3U};
+    char* args[numberOfArgs];
+    char appName[] = "./foo";
+    char option[] = "--unique-roudi-id";
+    char wrongValue[] = "4242";
```

Comment:
c&p error. I guess this should be only `value`

Reply:
Done